### PR TITLE
[Oracle Module] Improve connection documentation

### LINKED ISSUE #30949: [Oracle Module] Improve connection documentation
### Problem

- Oracle module’s documentation is confusing and not updated. Currently, it states that Oracle client must be installed as a pre-requisite which is not true. If you have OCI Oracle server which comes with these libraries pre-installed, you don't need a separate client installation, all you need is just the correct configuration settings. Since in this case, Oracle metricbeat module is tied to systemD running on Linux so the trick is to create a systemD configuration file that is pulled when metricbeat starts. This configuration file would provide environment variables that will help the Oracle module in finding the required libraries. It can take new customers weeks to figure out and troubleshoot.
### Solution

- Connectivity to Oracle can be facilitated in two ways either by using official Oracle libraries or by using a JDBC driver. These two approaches must be clearly spelled out in the documentation, whether they are supported or not and what libraries and settings must be configured, and where. Clear workflows and steps must be documented in a blog or a documentation page.

### User Story

- As an Oracle DBA and first-time Elastic user, I would like Elastic documentation to handhold me and clearly provide configurations for different Oracle setups and connectivity options so that I could understand how Oracle connects with Elastic Cloud and doesn’t have to waste time in experimentation and troubleshooting.

### PR DESCRIPTION
**Pull request by yug-elastic**
- Docs

## What does this PR do?

This PR improves the connection documentation for Connectivity to Oracle Database using OCI instant client for standalone Oracle Database installation as well as Dockerized Oracle Database.

## Why is it important?

Oracle module’s documentation is confusing and not updated. It can take new customers weeks to figure out and troubleshoot.

## Checklist

~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #30949

### CODE REVIEW

**Comment by lalit-satapathy** [comment]
@agithomas please take a look at this PR.

File: metricbeat/docs/modules/oracle.asciidoc [hunk 1]
```diff
@@ -28,14 +26,35 @@ image::./images/metricbeat-oracle-overview.png[]
...
 
-The OCI library install few Client Shared Libraries that must be referenced on the machine where Metricbeat is installed. The environment variable `LD_LIBRARY_PATH` is used to set the reference required by the Metricbeat library. For example, let's say you have downloaded the library into your `Downloads` folder, you can follow the steps below to install the ZIP library on `/usr/lib` (assuming you have `wget` and a zip unpacker):
+*Oracle Database Connection Pre-requisites*
```
Comment (agithomas) [inline metricbeat/docs/modules/oracle.asciidoc:30]: Solution ( https://github.com/elastic/beats/issues/30949)  Says: 
`Connectivity to Oracle can be facilitated in two ways either by using official Oracle libraries or by using a JDBC driver. These two approaches must be clearly spelled out in the documentation, whether they are supported or not and what libraries and settings must be configured, and where. `

So, kindly include the approaches that are supported and what isn't.

Comment (yug-elastic) [reply metricbeat/docs/modules/oracle.asciidoc:30]: Right, thank you. We started [this discussion](https://github.com/elastic/beats/issues/30949#issuecomment-1110761183) over the [PRD for the task](https://github.com/elastic/beats/issues/30949) for the same.

As mentioned there, based on our analysis, it won't be feasible to facilitate the connection to Oracle using JDBC. Should we mention the same or does anyone from the team has any difference of opinion for it?

Comment (yug-elastic) [reply metricbeat/docs/modules/oracle.asciidoc:32]: It's updated as discussed over the call. Please re-review the PR at your convenience.

**Review by agithomas — APPROVED** [review]

**Review by mtojek — APPROVED** [review]
LGTM

**Pull request merged**